src/Report/JunitReporter.php

```diff
@@ -0,0 +1,145 @@
+        }
+
+        if ($reportSummary->getTime()) {
+            $testsuite->setAttribute(
+                'time',
+                sprintf(
+                    '%.3f',
+                    $reportSummary->getTime() / 1000
+                )
+            );
+        }
+
+        $dom->formatOutput = true;
+
+        return $dom->saveXML();
+    }
+
+    private function createSuccessTestCase(\DOMDocument $dom, \DOMElement $testsuite)
+    {
+        $testcase = $dom->createElement('testcase');
+        $testcase->setAttribute('name', 'All OK');
+        $testcase->setAttribute('assertions', 1);
+
+        $testsuite->appendChild($testcase);
+        $testsuite->setAttribute('tests', 1);
+        $testsuite->setAttribute('assertions', 1);
+        $testsuite->setAttribute('failures', 0);
+        $testsuite->setAttribute('errors', 0);
+    }
+
+    /**
+     * @param \DOMDocument  $dom
+     * @param \DOMElement   $testsuite
+     * @param ReportSummary $reportSummary
+     */
+    private function createFailedTestCases(\DOMDocument $dom, \DOMElement $testsuite, ReportSummary $reportSummary)
+    {
+        $assertionsCount = 0;
+        foreach ($reportSummary->getChanged() as $file => $fixResult) {
+            $testcase = $this->createFailedTestCase(
+                $dom,
+                $file,
+                $fixResult,
+                $reportSummary->shouldAddAppliedFixers()
+            );
+            $testsuite->appendChild($testcase);
+            $assertionsCount += (int) $testcase->getAttribute('assertions');
+        }
+
+        $testsuite->setAttribute('tests', count($reportSummary->getChanged()));
+        $testsuite->setAttribute('assertions', $assertionsCount);
+        $testsuite->setAttribute('failures', $assertionsCount);
+        $testsuite->setAttribute('errors', 0);
+    }
+
+    /**
+     * @param \DOMDocument $dom
+     * @param string       $file
+     * @param array        $fixResult
+     * @param bool         $addAppliedFixers
+     *
+     * @return \DOMElement
+     */
+    private function createFailedTestCase(\DOMDocument $dom, $file, array $fixResult, $addAppliedFixers)
+    {
+        $appliedFixersCount = count($fixResult['appliedFixers']);
+
+        $testName = str_replace('.', '_DOT_', preg_replace('@\.'.pathinfo($file, PATHINFO_EXTENSION).'$@', '', $file));
+
+        $testcase = $dom->createElement('testcase');
+        $testcase->setAttribute('name', $testName);
+        $testcase->setAttribute('class', $testName);
+        $testcase->setAttribute('file', $file);
+        $testcase->setAttribute('assertions', $appliedFixersCount);
+
+        if ($addAppliedFixers) {
+            foreach ($fixResult['appliedFixers'] as $appliedFixer) {
+                $failure = $dom->createElement('failure');
+                $failure->setAttribute('type', $appliedFixer);
+                $appliedFixerReadable = ucfirst(str_replace('_', ' ', $appliedFixer));
```

Comment:
I would use name of rule as is, without any modifications.
in case of many rules violated it's nice to have the list of them instead of some modified strings